contracts/implementation/PoolFactory.sol

```diff
@@ -80,16 +86,34 @@ contract PoolFactory is IPoolFactory, Ownable {
     }
 
     /**
-     * @notice Deploy a leveraged pool with given parameters
+     * @notice Deploy a leveraged pool and its committer/pool tokens with given parameters
      * @param deploymentParameters Deployment parameters of the market. Some may be reconfigurable
      * @return Address of the created pool
      */
-    function deployPool(PoolDeployment calldata deploymentParameters) external override onlyGov returns (address) {
+    function deployPool(PoolDeployment calldata deploymentParameters) external override returns (address) {
         address _poolKeeper = address(poolKeeper);
         require(_poolKeeper != address(0), "PoolKeeper not set");
-        require(address(poolCommitterDeployer) != address(0), "PoolCommitterDeployer not set");
+        require(
+            IOracleWrapper(deploymentParameters.oracleWrapper).owner() == msg.sender,
+            "Deployer must be oracle wrapper owner"
+        );
+
+        bytes32 uniquePoolHash = keccak256(
+            abi.encode(
+                deploymentParameters.leverageAmount,
+                deploymentParameters.quoteToken,
+                deploymentParameters.oracleWrapper,
+                deploymentParameters.updateInterval,
+                deploymentParameters.frontRunningInterval
```

Comment:
> With that said however, you also define pool tokens to be unique based on the leverage amount, quote token and oracle wrapper -- meaning that someone wanting to deploy pools with the same first three params but different update/frontrunning intervals wouldn't be able to.

I agree with this.

However, I think having `frontRunningInterval` as a unique parameter is perfectly fine. It doesn't have a negative impact, but also gives us the potential for updates in the future. Fragmenting liquidity wouldn't be too much of a concern, since we would only change the frontrunning interval if there was something wrong with the previous one.

We definitely want `updateInterval` to be a unique parameter, in my opinion, because it has been mentioned in the past that having different update intervals of the same leverage/oracle could likely have a usecase.